miasm2/core/asmblock.py

```diff
@@ -764,33 +904,35 @@ def sanity_check(self):
         """
 
         if len(self._pendings) != 0:
-            raise RuntimeError("Some blocks are missing: %s" % map(str,
-                                                                   self._pendings.keys()))
+            raise RuntimeError("Some blocks are missing: %s" % map(
+                str,
+                self._pendings.keys()
+            ))
 
         next_edges = {edge: constraint
                       for edge, constraint in self.edges2constraint.iteritems()
                       if constraint == AsmConstraint.c_next}
 
-        for block in self._nodes:
+        for label in self._nodes:
             # No next constraint to self
-            if (block, block) in next_edges:
+            if (label, label) in next_edges:
                 raise RuntimeError('Bad constraint: self in next')
 
             # No multiple next constraint to same block
-            pred_next = list(pblock
-                             for (pblock, dblock) in next_edges
-                             if dblock == block)
+            pred_next = list(plabel
+                             for (plabel, dlabel) in next_edges
+                             if dlabel == label)
 
             if len(pred_next) > 1:
                 raise RuntimeError("Too many next constraints for bloc %r"
-                                   "(%s)" % (block.label,
-                                             map(lambda x: x.label, pred_next)))
+                                   "(%s)" % (label,
+                                             pred_next))
 
```

Comment:
Regarding the situation described above, we may want to temporarily consider nodes without corresponding block as a sanity error

Reply:
done